src/node/wrapper.ts

```diff
@@ -253,21 +268,17 @@ export class ParentProcess extends Process {
   private async relaunch(): Promise<void> {
     this.disposeChild()
     try {
-      await this.start()
+      this.started = this._start()
+      await this.started
     } catch (error) {
       this.logger.error(error.message)
       this.exit(typeof error.code === "number" ? error.code : 1)
     }
   }
 
-  public start(): Promise<void> {
-    // If we have a process then we've already bound this.
-    if (!this.child) {
-      process.on("SIGUSR1", async () => {
-        this.logger.info("Received SIGUSR1; hotswapping")
-        this.relaunch()
-      })
-    }
+  public start(args: DefaultedArgs): Promise<void> {
+    // Store for relaunches.
+    this.args = args
```

Comment:
Why not put this in the constructor?

Reply:
The only reason is because it's constructed in this file where we don't have args yet.

Reply:
Could definitely move the `new` parts into `entry.ts` and the `process.stdout.isTTY` and `uncaughtException` bits as well.

Reply:
Ah I gotchu, fine to leave as is then.